### ISSUE TITLE
Error: MDX doesn't support using "default" as a named export

### ISSUE THREAD

**Issue by dingyi** [ChristopherBiscardi/gatsby-mdx#222]
**Describe the bug**
The starter example was working perfectly. But when I added it to my exciting project, it shows this error:

error UNHANDLED REJECTION
  Error: MDX doesn't support using "default" as a named export, use "export default" statement instead.
  For example, instead of:
  export { Layout as default }
  use:
  export default Layout

**Screenshots**

<img width="685" alt="screen shot 2018-12-31 at 22 38 07" src="https://user-images.githubusercontent.com/2486/50562181-caa50a00-0d4c-11e9-86b0-29edcaeb69cf.png">

I don't know what's wrong, all my mdx file are correct. Any suggestions?

**Comment by johno** [comment]
Hey @dingyi do you mind posting the exports from your MDX file that's erroring?

**Comment by dingyi** [comment]
I found what's wrong, when I disabled these configures from gatsby-config.js

```
{
    resolve: `gatsby-mdx`,
    options: {
      gatsbyRemarkPlugins: [
      {
        resolve: "gatsby-remark-images",
        options: {
          maxWidth: 1035,
          sizeByPixelDensity: true
        }
      }
      ]
    }
  },
```
It works, seems everything is fine. And I just tested it, any gatsby-mdx configures will cause this error.

I Googled it and only found one similar issues here: https://github.com/mdx-js/mdx/issues/347

No idea where this error comes from.

**Comment by dashed** [comment]
I'm running into a similar issue when following some docs on MDX layouts: https://gatsby-mdx.netlify.com/guides/using-mdx-layouts

------

post.mdx:
```mdx
import PostTemplate from '../components/post_template';
export default PostTemplate;


# My MDX
some content
```

The error:

```
ℹ ｢wdm｣: Compiling...
error UNHANDLED REJECTION


  SyntaxError: unknown: Only one default export allowed per module. (4:0)
    2 | export default PostTemplate;
    3 |
  > 4 | export default class MDXContent extends React  .Component {
      | ^
    5 |   constructor(props) {
    6 |     super(props)
    7 |     this.layout = null
```

**Comment by ChristopherBiscardi** [comment]
@dashed include a newline between the import and the export.

```js
import PostTemplate from '../components/post_template';

export default PostTemplate;
```

I think it's related to how the import/exports work in mdx.

**Comment by johno** [comment]
Yeah, this is an [MDX bug](https://github.com/mdx-js/mdx/issues/340) that's being worked on.
